docs/docs/optimizing-performance.md

```diff
@@ -22,23 +40,63 @@ shouldComponentUpdate(nextProps, nextState) {
 }
 ```
 
-If you know that in some situations your component doesn't need to update, you can return `false` from `shouldComponentUpdate` instead, to skip the slow DOM update. For example, if the only way your component ever changes is when the `state.count` variable changes, you could write:
+If you know that in some situations your component doesn't need to update, you can return `false` from `shouldComponentUpdate` instead, to skip the whole rendering process, including calling `render()` on this component and below. For example, if the only way your component ever changes is when the `props.color` or the `state.count` variable changes, you could have `shouldComponentUpdate` check that:
 
 ```javascript
-shouldComponentUpdate(nextProps, nextState) {
-  return this.state.count !== nextState.count;
+class CounterButton extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {count: 1};
+  }
+
+  shouldComponentUpdate(nextProps, nextState) {
+    if (this.props.color !== nextProps.color) {
+      return true;
+    }
+    if (this.state.count !== nextState.count) {
+      return true;
+    }
+    return false;
+  }
+
+  render() {
+    <button
+      color={this.props.color}
+      onClick={() => this.setState(state => {count: state.count + 1})}>
```

Comment:
Pitfall: this won't work because `state => { ... }` won't return anything.
You want to use `state => ({ ... })`  instead which produces an object expression.